src/Fantomas.Tests/IfThenElseTests.fs

```diff
@@ -915,4 +915,157 @@ let ``comment after then in if/then, 730`` () =
     |> should equal """
 if true then // comment
     ()
-"""
\ No newline at end of file
+"""
+
+[<Test>]
+let ``keep then in same line as if, 825`` () =
+
+    let c = { config with
+               KeepIfThenInSameLine = true
+               PageWidth = 80
+               MaxIfThenElseShortWidth = 0 }
+
+    formatSourceString false """type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        if balanceAtTheMomentOfSending < valueToSend then
+            invalidArg "balanceAtTheMomentOfSending"
+                "some very very long error message"
+        if valueToSend <= 0m then
+            invalidArg "valueToSend" "Amount has to be above zero"
+"""  c
+    |> prepend newline
+    |> should equal """
+type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        if balanceAtTheMomentOfSending < valueToSend then
+            invalidArg "balanceAtTheMomentOfSending"
+                "some very very long error message"
+        if valueToSend <= 0m then
+            invalidArg "valueToSend" "Amount has to be above zero"
+"""
+
+[<Test>]
+let ``keep then in same line as if with comments, 825`` () =
+
+    let c = { config with
+               KeepIfThenInSameLine = true
+               PageWidth = 80
+               MaxIfThenElseShortWidth = 0 }
+
+    formatSourceString false """type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        // comment
+        if balanceAtTheMomentOfSending < valueToSend then // comment
+            invalidArg "balanceAtTheMomentOfSending" // comment
+                "some very very long error message" // comment
+        if valueToSend <= 0m then // comment
+            invalidArg "valueToSend" "Amount has to be above zero" // comment
+"""  c
+    |> prepend newline
+    |> should equal """
+type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        // comment
+        if balanceAtTheMomentOfSending < valueToSend then // comment
+            invalidArg "balanceAtTheMomentOfSending"  // comment
```

Comment:
I noticed an additional whitespace is added between the code and the comment. I am not sure if this is desired behaviour and it seems to be unrelated to the changes I have introduced.

Reply:
I would imagine this is unrelated. We can follow this up in a separate issue.